### PR TITLE
beacon/blsync: proceed with empty finalized hash if proof is not expected soon

### PR DESCRIPTION
**Pull request by zsfelfoldi**
This PR fixes an issue that happens when the chain either cannot finalize or is new and does not have a finality proof yet. Now if a finality proof is not expected soon then blsync proceeds without a finalized block hash.

### CODE REVIEW

File: beacon/blsync/block_sync.go [hunk 1]
```diff
@@ -117,15 +118,32 @@ func (s *beaconBlockSync) updateEventFeed() {
...
+			}
+			if he == fe+1 {
+				if parent, ok := s.recentBlocks.Get(head.Header.ParentRoot); !ok ||
+					parent.Slot()/params.EpochLength == fe {
+					return // head is at first slot of next epoch, wait for finality update
+					//TODO trt to fetch finality update directly if subscription does not deliver
+				}
+
+			}
+		}
```
Comment (fjl) [inline beacon/blsync/block_sync.go:144]:
```suggestion
		he := head.Header.Epoch()
		fe := finality.Attested.Header.Epoch()
		switch {
		case he == fe:
			finalizedHash = finality.Finalized.PayloadHeader.BlockHash()
		case he < fe:
			return
		case he == fe+1:
			parent, ok := s.recentBlocks.Get(head.Header.ParentRoot)	
			if !ok || parent.Slot()/params.EpochLength == fe {
				return // head is at first slot of next epoch, wait for finality update
				//TODO trt to fetch finality update directly if subscription does not deliver
			}
		}
```

Comment (zsfelfoldi) [reply beacon/blsync/block_sync.go]: Nicer indeed, thanks. Committed and rebased on master.

File: beacon/blsync/block_sync.go [hunk 2]
```diff
@@ -117,15 +118,31 @@ func (s *beaconBlockSync) updateEventFeed() {
...
+		switch {
+		case he == fe:
+			finalizedHash = finality.Finalized.PayloadHeader.BlockHash()
+		case he < fe:
+			return
+		case he == fe+1:
+			parent, ok := s.recentBlocks.Get(head.Header.ParentRoot)
+			if !ok || parent.Slot()/params.EpochLength == fe {
+				return // head is at first slot of next epoch, wait for finality update
+				//TODO trt to fetch finality update directly if subscription does not deliver
```
Comment (lightclient) [inline beacon/blsync/block_sync.go:141]:
```suggestion
				//TODO: try to fetch finality update directly if subscription does not deliver
```

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**